src/transformers/models/swin/modeling_swin.py

```diff
@@ -167,12 +193,17 @@ def __init__(self, input_resolution, dim, norm_layer=nn.LayerNorm):
         self.reduction = nn.Linear(4 * dim, 2 * dim, bias=False)
         self.norm = norm_layer(4 * dim)
 
-    def forward(self, input_feature):
-        height, width = self.input_resolution
+    def forward(self, input_feature, height, width):
+        # height, width = self.input_resolution
```

Comment:
Needs to be cleaned up.

Reply:
now all functions that needs to know the spatial dimension, a tuple `input_dimensions` is always passed